### PR TITLE
fix(pg-cdc): fix postgres.publication_exist, found in issues#18628

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As titled

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by hzxa21 — APPROVED** [review]

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java [hunk 1]
```diff
@@ -62,8 +62,10 @@ public static void createPostgresPublicationIfNeeded(
                             ValidatorUtils.getSql("postgres.publication_exist"))) {
                 stmt.setString(1, pubName);
                 var res = stmt.executeQuery();
+                // Note: the value returned here is `pubviaroot`, If there's more than one row, the
```
Comment (StrikeW) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java:65]: The `pubviaroot` only exist in PG version >= 13, we should use a more general sql to check the existence of the publication.

Comment (KeXiangWang) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java:65]: 🤔OKK, will use another PR to refactor this.

Comment (StrikeW) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java:65]: I think we can revert back to use `count(*)` to check the existence.

Comment (KeXiangWang) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java:65]: Partitioned table support relies on `pubviaroot`. Simply reverting will break the partitioned table support.

**Comment by cyliu0** [comment]
https://buildkite.com/risingwave-test/sysbench-cdc/builds/681#01921e48-8a30-405c-801c-9de0e1140a87

The test passed @KeXiangWang

**Comment by KeXiangWang** [comment]
Thanks @cyliu0

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java (hunk 1 above)
Comment (KeXiangWang) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/DbzSourceUtils.java:65]: fixed in https://github.com/risingwavelabs/risingwave/pull/18648

**Pull request merged**